### ISSUE TITLE
Program hangs at 'Running git clone for libxml2...'

### ISSUE THREAD

**Issue by AsianAmericanManAD1979** [m-ab-s/media-autobuild_suite#1551]
How to fix this?  Try compiling ffmpeg without libxml2?

        Starting 64bit compilation of all tools
Imported options from ffmpeg_options.txt
License: nonfree
Imported options from mpv_options.txt

        Starting 64bit compilation of global tools
├ Running rustup_update...
├ Running set_default_toolchain...
  Running git clone for libxml2...

**Comment by 1480c1** [comment]
Can you open mintty from the root folder and run
```bash
source /build/media-suite_helper.sh
do_vcs "https://gitlab.gnome.org/GNOME/libxml2.git"
```
and paste the output here?

**Comment by AsianAmericanManAD1979** [comment]
I use Avast Free Antivirus.  It has locked away a file named scdaemon.exe in C:\mingw\media-autobuild_suite-master\msys64\usr\lib\gnupg, because of a threat it has identified as IDP.Generic.

I have turn off File Shield and Behavior Shield in Avast Free Antivirus, so that it can no longer interfere or interrupt the ffmpeg compiling process.

I have commented out "--enable-libxml2" in ffmpeg_options.txt.

As of now, the libraries required by ffmpeg are being compiled on my computer.

I'll post an update when the compiling is over.

After this compilation is over, I may try to recompile with --enable-libxml2 on.

I'll post every relevant piece of information, once I'm done with compilation(s).

**It looks like every antivirus program should be turned off before running media-autobuild_suite to compile all of ffmpeg dependencies and ffmpeg; otherwise, the antivirus program would interfere.  (This must be stressed.)**

The following is my current media-autobuild_suite terminal output.



        Starting 64bit compilation of all tools
Imported options from ffmpeg_options.txt
License: nonfree
Imported options from mpv_options.txt

        Starting 64bit compilation of global tools
├ Running rustup_update...
├ Running set_default_toolchain...
  Running git clone for freetype2...
┌ freetype2 git  ...................................... [Recently updated]
├ Running autogen...
├ Running configure...
├ Running make...
├ Running install...
├ Stripping...
└ freetype2-git  ............................................... [Updated]
  Running git clone for fontconfig...
┌ fontconfig git  ..................................... [Recently updated]
├ Running autogen...
├ Running configure...
├ Running make...
├ Running install...
├ Stripping...
└ fontconfig-git  .............................................. [Updated]
  Running git clone for harfbuzz...
┌ harfbuzz git  ....................................... [Recently updated]
├ Installing ragel... done
├ Running autogen...
├ Running configure...
├ Running make...

**Comment by 1480c1** [comment]
yep. windows defender takes about 50% of my cpu while compiling

**Comment by gapbl4** [comment]
is adding suite folder to exceptions a good enough measure to have build go without issues related to defender?

**Comment by 1480c1** [comment]
yes, I just added the suite as an exception

**Comment by pat357** [comment]
How can you add a folder to exclude in Windows Defender ?
I'm going though all the menus, but I see  no option to exclude a folder....

When I'm in **the main menu from defender** I see the following **sub-menus**:
(I have a dutch Win10, so some words might be in Dutch language)

- Virus and Threads Security
- Account security
- Firewall and network security
- Apps and Browser Management
- Apparaat Beveiliging ( =~ Device security ??)
- Apparaat prestaties and status {=~ performance and status)
- Family options

Where should I start looking for a possibility to exclude a folder ?

**Comment by gapbl4** [comment]
![](https://i.imgur.com/0mRCnlJ.gif)

**Comment by pat357** [comment]
@gapbl4 Thank you so much !!
I successfully added my m-ab-s folder to the exclusions.

I wonder how you created an animated post here ....  is it by attaching an animated gif or something like that ?

**Comment by 1480c1** [comment]
you can attach images by drag and drop or
```markdown
![pat357](https://avatars2.githubusercontent.com/u/13542862?s=460&v=4)
```
which leads to 
![pat357](https://avatars2.githubusercontent.com/u/13542862?s=460&v=4)

**Comment by gapbl4** [comment]
@pat357 it's by using [https://github.com/ShareX/ShareX](url) that uploads the gif to imgur or other hosting of your choice. I just dragged the resulting image link from browser tab to the input box.

**Comment by AsianAmericanManAD1979** [comment]
The first ffmpeg compilation attempt without --enable-libxml2 failed, because dav1d compilation failed.

It seems that libxml2 is used by one or more of the libraries that ffmpeg is dependent on—so, if you turn off --enable-libxml2, ffmpeg compilation will probably fail because compiling a library that ffmpeg needs will fail.  dav1d certainly needs libxml2 for successful compilation.

**Very likely, compiling will fail, if you don't turn off your antivirus program, or exclude your ffmpeg compilation directory from antivirus monitoring—as noted in the previous posts by others.  Make sure that --enable-libxml2 not commented out in ffmpeg_options.txt.
[ab-suite.configure.log](https://github.com/m-ab-s/media-autobuild_suite/files/4175593/ab-suite.configure.log)
[entire terminal output.txt](https://github.com/m-ab-s/media-autobuild_suite/files/4175594/entire.terminal.output.txt)
[logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/4175595/logs.zip)


**


The second compilation attempt with --enable-libxml2 failed, because running configure failed while compiling the ffmpeg main program.  It says " ERROR: libmysofa not found".

In this post, I'm attaching all the relevant outputs of the second compilation attempt.


I'll think about why the latest compilation attempt has failed.

I might consider compiling without libmysofa.

I'll review the logs and try to figure out what went wrong.

In the ffmpeg main program compilation section of the terminal output, it says, "Until SVT-AV1 issues a fix, libaom & libopencore-amrwb must be disabled while libsvtav1 is enabled....".  In my ffmpeg_options.txt, libaom, libopencore-amrwb, and libsvtav1 are all enabled:  I didn't know what they were, so I did not disable them.  That might have caused the ffmpeg main program configure failure.  I might make another attempt at compiling with that requirement satisfied.

I'll review the logs, and think about what to do next, as soon as I can.

I'll post an update, after I figure out what to try next, and try it.


Any comment on how to fix this will be greatly appreciated.

**Comment by AsianAmericanManAD1979** [comment]
I found some earlier threads related to the SVT-AV1 issue.  I'm posting here as references for those who come across this thread.
[https://github.com/m-ab-s/media-autobuild_suite/issues/1364](https://github.com/m-ab-s/media-autobuild_suite/issues/1364)
[https://github.com/m-ab-s/media-autobuild_suite/pull/1365](https://github.com/m-ab-s/media-autobuild_suite/pull/1365)

I'll keep working on resolving this issue, and I'll keep posting relevant and useful updates until I succeed at compiling ffmpeg.

**Comment by AsianAmericanManAD1979** [comment]
I attempted another compilation with --enable-libsvtav1 commented out in ffmpeg_options.txt.

The compilation failed with the same error, "libmysofa not found".
(error files here.)
[entire terminal output.txt](https://github.com/m-ab-s/media-autobuild_suite/files/4176914/entire.terminal.output.txt)
[ab-suite.configure.log](https://github.com/m-ab-s/media-autobuild_suite/files/4176915/ab-suite.configure.log)
[logs.zip](https://github.com/m-ab-s/media-autobuild_suite/files/4176916/logs.zip)




C:\mingw\media-autobuild_suite-master\build\libmysofa-git folder existed with files in them.

The terminal output showed that libmysofa was present and up-to-date.

I attempted another compilation with --enable-libmysofa commented out in ffmpeg_options.txt.

The compilation succeeded.  Everything compiled, and the compilation-statuses terminal closed on its own.

In the ffmpeg main program compilation section of the terminal output, it said, "patch unexpectedly ends in middle of line".  Although that did not stop ffmpeg from being compiled, it probably meant there was an issue in the ffmpeg source code.

The compiled static ffmpeg executables, in C:\mingw\media-autobuild_suite-master\build\ffmpeg-git\build-static-64bit, were fairly big in file sizes.
ffmpeg.exe 110,229,504
ffmpeg_g.exe 133,798,015
ffplay.exe 110,117,376
ffplay_g.exe 133,674,395
ffprobe.exe 110,146,560
ffprobe_g.exe 133,706,227
(I didn't know what the ones with the _g suffix were for.)

When I tried to run ffmpeg, Windows produced the following error.

"The program can't start because libopenh264.dll is missing from your computer.  Try reinstalling the program to fix this problem."

Apparently, the compiled ffmpeg executable was not completely static.

After doing local file search, I found out that libopenh264.dll was present in two different locations.  C:\mingw\media-autobuild_suite-master\local64\bin-video, and C:\mingw\media-autobuild_suite-master\msys64\mingw64\bin.

C:\mingw\media-autobuild_suite-master\msys64\mingw64\bin didn't have the compiled ffmpeg executable in it.  C:\mingw\media-autobuild_suite-master\local64\bin-video did.

C:\mingw\media-autobuild_suite-master\local64\bin-video seemed to be the output folder for all the compiled dynamic-library and executable files.

In Windows Terminal, I navigated to C:\mingw\media-autobuild_suite-master\local64\bin-video does, then executed ffmpeg.

ffmpeg ran without an error.

I have done a basic test with the compiled ffmpeg executable, by executing the compiled ffmpeg executable with the following parameters.  No error, and the output video seemed ok.  However, the bit rate of the output audio in the video was 133kbps, not 128kbps—not exact.

ffmpeg -y -i "Dog - 15305 - original.mp4" -r 30 -c:v libx264 -preset slow -crf 18 -pix_fmt yuv420p -c:a libfdk_aac -b:a 128k -ar 48000 -ac 2 -shortest -movflags +faststart "Dog - 15305.mp4"


In summary, here's what I did to successfully compile ffmpeg with Media Autobuild Suite.

-Turned off my antivirus program during compilation.
-Made sure that --enable-libxml2 was on.
-Commented out --enable-libsvtav1 in ffmpeg_options.txt.
-Commented out --enable-libmysofa in ffmpeg_options.txt.

Here are my ffmpeg_options.txt and mpv_options.txt files that resulted in successful compilation of ffmpeg.  (I didn't know what mpv was, so I selected compiling that during the initial compilation options configuring.)
[ffmpeg_options.txt](https://github.com/m-ab-s/media-autobuild_suite/files/4176912/ffmpeg_options.txt)
[mpv_options.txt](https://github.com/m-ab-s/media-autobuild_suite/files/4176913/mpv_options.txt)


I'm closing this issue now.

It was a nice little computing adventure!
